### ISSUE TITLE
Consider pulling in imports via an explicit command

### ISSUE THREAD

**Issue by bholley** [mozilla/cargo-vet#391]
Right now, we re-check the import set every time somebody invokes a non-`--locked` cargo-vet command. If There's a new import that would allow us to remove an exemption, we automatically pull it in and drop the exemption.

Experience suggests that this is confusing for rank-and-file developers. They're usually in the process of auditing some dependency they added, and then some stuff for an unrelated crate appears in the same commit. Given all the ceremony of sign-off, it's a particularly bad moment for the tool to make mysterious changes without explanation.

I think we should switch to a model where cargo-vet simply prints a message after the "Vetting Succeeded" that says something like: `New audits are available for import — Please invoke cargo vet fetch-imports.` Or something like that.

**Comment by bholley** [comment]
@mystor @afranchuk @alexcrichton @jameysharp WDYT?

**Comment by bholley** [comment]
I guess we already have `fetch-imports`, which fetches all the imports for a project's crates rather than just the crates for which an exemption can be removed. We could perhaps change the behavior and add an `--all` flag for the old behavior, or add a different top-level command.

**Comment by alexcrichton** [comment]
I think printing a message is pretty reasonable, effectively otherwise not updating the exemptions list automatically unless explicitly requested.

**Comment by afranchuk** [comment]
I like making it explicit and adding a message. I agree that it is surprising/confusing to see changes occur that a user didn't initiate themselves in some way.

**Comment by mystor** [comment]
This has some potential conflict with the desires from #262, where this automatic exemption removal behaviour was added. 

When we suggest an audit, we may suggest an audit built on top of an unimported base audit from a peer, and we definitely need to import that audit after you `certify` it in order to get `vet` passing reliably, so doing an import implicitly in some cases is definitely something which needs to happen. That being said, we can perhaps be a bit more clever about when we do an implicit import and remove exemptions, by limiting exemption removal to only fully local codepaths, or the named crate when doing a `cargo vet certify`.

One of the reasons we didn't do that originally is that because of `dependency-criteria`, we could only minimize exemptions on fully-vetting trees, meaning that if you certified a crate, we couldn't remove any associated exemptions until the tree was fully vetted, and by that point (after multiple `certify` calls) we didn't know what crates you had certified anymore, so we just tried to remove every exemption.

Now that dependency-criteria on audits is gone, we can theoretically perform imports and exemption minimization for each crate individually (though this has not been implemented), which might make it easier to limit new imports to only failing trees (i.e. no exemption removals), unless `certify` is being run for a specific crate.
